nerdery-ecommerce/src/products/products.service.ts

```diff
@@ -115,5 +124,117 @@ export class ProductsService {
     });
   }
 
+  async create(input: CreateProductInput) {
+    const { categoryId, ...rest } = input;
+
+    if (!categoryId) {
+      throw new BadRequestException('Category is required');
+    }
+
+    if (!await this.categoriesService.doesCategoryExist(categoryId)) {
+      throw new BadRequestException('Category does not exist');
+    }
+
+
+    return await this.prisma.product.create({
+      data: {
+        ...rest,
+        category: {
+          connect: {
+            id: categoryId
+          }
+        },
+        //Default values
+        thumbnailUrl: '',
```

Comment:
Refractor, avoid passing optional params if Prisma Schema has these as Default in the database